tiscale_connector.py

```diff
@@ -312,19 +312,21 @@ def _make_rest_call(
 
     def _get_file_dict(self, param, action_result):
 
-        vault_id = param['vault_id']
+        vault_id = param[TISCALE_JSON_VAULT_ID]
 
         filename = param.get('file_name')
         if not filename:
             filename = vault_id
 
         try:
-            if (hasattr(Vault, 'get_file_path')):
-                payload = open(Vault.get_file_path(vault_id), 'rb')
-            else:
-                payload = open(
-                    Vault.get_vault_file(vault_id),
-                    'rb')  # pylint: disable=E1101
+            success, msg, files_array = ph_rules.vault_info(vault_id=vault_id)
+            if not success:
+                return (action_result.set_status(phantom.APP_ERROR,
+                        f'Unable to get Vault item details. Error Details: {msg}'),
+                        None)
+            file_data = list(files_array)[0]
```

Comment:
```suggestion
            file_data = next(iter(files_array))
```
to avoid generating the whole list